### PR TITLE
clients(viewer): upgrade to firebase 9

### PR DESCRIPTION
**Pull request by connorjclark**
* firebase 4.1.2 -> 9.0.2
* get firebase from npm instead of cdn–saves additional origin connect, treeshakes (saves ~25 kB), and gives us easier types
* some analytics features enabled in firebase, idk, didn't look into that
* did some refactoring (deleting unused code, using async/await for readability)

### CODE REVIEW

File: lighthouse-viewer/app/src/firebase-auth.js [hunk 1]
```diff
@@ -49,42 +47,36 @@ export class FirebaseAuth {
...
+  async signIn() {
+    const result = await signInWithPopup(this._auth, this._provider);
+    const credential = GithubAuthProvider.credentialFromResult(result);
+    if (!credential || !credential.accessToken) throw new Error('unexpected credential');
 
-  /**
-   * Signs the user out.
-   * @return {Promise<void>}
-   */
-  signOut() {
```
Comment (patrickhulce) [inline lighthouse-viewer/app/src/firebase-auth.js:84]: unused?

**Review by patrickhulce — APPROVED** [review]

File: lighthouse-viewer/app/src/firebase-auth.js (hunk 1 above)
Comment (connorjclark) [reply lighthouse-viewer/app/src/firebase-auth.js:84]: yes

**Comment by brendankenny** [comment]
> * some analytics features enabled in firebase, idk, didn't look into that

what does this mean?

**Comment by connorjclark** [comment]
w/e `measurementId` (new property, but needs newer sdk) enabled. https://firebase.google.com/docs/analytics/get-started?platform=web

**Comment by brendankenny** [comment]
seems like we should turn it off?

**Comment by connorjclark** [comment]
> some analytics features enabled in firebase, idk, didn't look into that

OK, looked into it. it's not enabled. (`firebase/analytics` is not used)

**Pull request merged**